mlir/test/Dialect/Linalg/transform-lower-pack.mlir

```diff
@@ -96,6 +96,34 @@ module attributes {transform.with_named_sequence} {
 
 // -----
 
+// This is same as pack_as_pad but since we explicitly added {allowInsertSliceLowering = false}, it should not
+// be lowered to insert_slice.
+// CHECK-LABEL: func.func @pack_disallowed_as_pad(
+// CHECK: %[[SRC:.+]]: tensor<129x47x16x16xf32>,
+// CHECK: %[[OUT:.+]]: tensor<1x1x1x1x136x64x16x16xf32>)
+func.func @pack_disallowed_as_pad(%arg0: tensor<129x47x16x16xf32>, %arg1: tensor<1x1x1x1x136x64x16x16xf32>) -> tensor<1x1x1x1x136x64x16x16xf32> {
+  %cst_0 = arith.constant 0.0 : f32
+  // tensor.pack is lowered to tensor.pad + tensor.expand_shape + tensor.insert_slice
+  //      CHECK: %[[PAD:.*]] = tensor.pad %[[SRC]] low[0, 0, 0, 0] high[7, 17, 0, 0]
+  //      CHECK:   : tensor<129x47x16x16xf32> to tensor<136x64x16x16xf32>
```

Comment:
If there's no `tensor.insert_slice`, how is the output actually generated?

Reply:
Without the tensor.insert_slice it just follows the normal lowering path, which is well tested by other lit tests. I think it is okay for this test to be a simple check that the option kicked in, and we can avoid having more tests to change if there is ever a change to the pattern implementation.

Reply:
I agree that it's unnecessary to generate `CHECK` lines for behavior already covered elsewhere. However, I see this test as a variation of `@pack_as_pad`, which produces:
```mlir
%pad = tensor.pad
%slice = tensor.insert_slice %pad
return %slice
```

What's the actual sequence here? I was hoping the lowerPack docs would clarify: https://github.com/llvm/llvm-project/blob/157d847ba737b4136aeb1d92912f549ea1c96d4c/mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h#L1122-L1124

However, the description is sparse and mentions reshape and transpose, which don’t seem to apply here. Is `PAD` the actual return value in this case?

Reply:
No it will be pad + reshape + transpose and return value is from transpose. Adding reshape and transpose kind-of lose its point because this unit test is not to test that it has reshape and transpose -that's for a dozen unit test above. This is very specifically focused on not having insert or extract slice. I'd prefer orthogonal test coverage if we must have it.

> Nit maybe, but the fact that you have tests in a NFC change is a bit surprising to me?

I tends to agree with @joker-eph that this doesn't need to come with such a unit test based on:
 - Optional attributes in other transform ops are not tested
 - This is NFC PR and doesn't change any existing